### PR TITLE
#863 - Added compulsory braces for if and else blocks (if multi-line)

### PR DESCRIPTION
**Pull request by codemanmeet**
Also - Updated coding conventions document. Fixed resulting check style errors.

### CODE REVIEW

File: docs/codingConventions.md [hunk 1]
```diff
@@ -9,6 +9,6 @@ We've tweaked a few of the rules for our needs:
 - Import order matches Eclipse's default (no priority given to Google-specific package names)
 - Such `__METHOD_NAMES__` are allowed
 - Line length is increased to 120
-- `if` and `else` blocks don't need to have braces. This is useful for guard clauses and automatically-generated equals methods.
+- `if` and `else` blocks don't need to have braces. This is useful for guard clauses and automatically-generated equals methods. Although, any `if` and `else` blocks that are multi-line require braces.
```
Comment (damithc) [inline docs/codingConventions.md]: Reverse the phrasing to put the rule before the exception: They should have braces unless it is a single line ...

Comment (codemanmeet) [inline docs/codingConventions.md]: Addressed!

**Comment by dariusf** [comment]
No problems here. Minor thing before merging: the two merges aren't necessary, since they happened at the beginning. Would it be possible to remove them? The easiest way is to cherry pick the rest onto master.

**Comment by codemanmeet** [comment]
Ok, I'll cherry pick the commit to master then.

**Comment by dariusf** [comment]
What I meant was to remove the merges from this branch by cherry-picking onto master, making that head the new `863-coding-convention-braces` branch, then merging that via the usual PR workflow. This is so we can undo the work done here if necessary.

I'll revert the changes to master and do that.

**Comment by codemanmeet** [comment]
Oh got it

**Pull request merged**